### DISCUSSION TITLE
Piping commands doesn't work

### DISCUSSION THREAD

**Discussion by thisismydesign** [community/community#25906] [Actions]
<p>My command is this (I need to turn the output of <code>aws</code> into a 1-liner so that I can make it an output of the step. Multi-line outputs are not supported).:</p>
<p><code>aws amplify start-job --app-id xxx --branch-name master --job-type RELEASE | jq -c</code></p>
<p>I get usage instructions back from <code>jq</code>, meaning it didn’t receive the data. It works locally and I can see the output of the <code>aws</code> command without the pipe.</p>
<p>What might be the issue?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/thisismydesign">@thisismydesign</a>,</p>
<p>Based on the <a href="https://stedolan.github.io/jq/manual/" rel="nofollow noopener">usage</a>:<br>
<code>--compact-output / -c:</code><br>
<code>By default, jq pretty-prints JSON output. Using this option will result in more compact output by instead putting each JSON object on a single line.</code></p>
<p>What’s the output of the command <code>without</code> the pipe? Is it json format?<br>
If it’s not, please try to use <code>command | xargs</code> instead, it will set the output to oneline.</p>
<pre><code class="lang-auto">aws amplify start-job --app-id xxx --branch-name master --job-type RELEASE | xargs
</code></pre>
<p>Thanks</p>
